cardano-cli/src/Cardano/CLI/Shelley/Run.hs

```diff
@@ -36,19 +36,36 @@ data ShelleyClientCmdError
   | ShelleyCmdKeyError !ShelleyKeyCmdError
   deriving Show
 
-renderShelleyClientCmdError :: ShelleyClientCmdError -> Text
-renderShelleyClientCmdError err =
-  case err of
-    ShelleyCmdAddressError addrCmdErr -> renderShelleyAddressCmdError addrCmdErr
-    ShelleyCmdGenesisError genesisCmdErr -> renderShelleyGenesisCmdError genesisCmdErr
-    ShelleyCmdGovernanceError govCmdErr -> renderShelleyGovernanceError govCmdErr
-    ShelleyCmdNodeError nodeCmdErr -> renderShelleyNodeCmdError nodeCmdErr
-    ShelleyCmdPoolError poolCmdErr -> renderShelleyPoolCmdError poolCmdErr
-    ShelleyCmdStakeAddressError stakeAddrCmdErr -> renderShelleyStakeAddressCmdError stakeAddrCmdErr
-    ShelleyCmdTextViewError txtViewErr -> renderShelleyTextViewFileError txtViewErr
-    ShelleyCmdTransactionError txErr -> renderShelleyTxCmdError txErr
-    ShelleyCmdQueryError queryErr -> renderShelleyQueryCmdError queryErr
-    ShelleyCmdKeyError keyErr -> renderShelleyKeyCmdError keyErr
+-- Identity monad is used to avoid boilerplate
+renderShelleyClientCmdError :: ShelleyCommand -> ShelleyClientCmdError -> Identity Text
```

Comment:
Why the identitiy monad?

Reply:
I used the `Identity` monad to avoid this boilerplate:
```
case err  of 
  ShelleyCmdADdressError addrCmdErr -> "Shelley command failed:" <> renderShelleyCommand cmd
                                                                                                               <> "  Error: "
                                                                                                               <> renderShelleyAddressCmdError addrCmdErr
  ShelleyCmdGenesisError genesisCmdErr -> "Shelley command failed:" <> renderShelleyCommand cmd
                                                                                                               <> "  Error: "
                                                                                                               <> renderShelleyGenesisCmdError addrCmdErr
```